packages/material-ui/test/integration/MenuList.test.js

```diff
@@ -6,7 +6,8 @@ import MenuItem from 'packages/material-ui/src/MenuItem';
 import { createMount } from 'packages/material-ui/src/test-utils';
 
 function assertMenuItemTabIndexed(wrapper, tabIndexed) {
-  const items = wrapper.find('MenuItem');
+  const items = wrapper.find('li[role="menuitem"]');
+  assert.strictEqual(items.length, 4, 'should find 4 items');
```

Comment:
@ryancogswell We avoid providing a fail message to the assertion. We just don't have the time to maintain them. Jest don't even allow it. What do you think of removing the message?

Reply:
Well, for rare cases we add the fail messages as it helps debug, it might be once of these cases.

Reply:
The only reason I had it is because the other assertions in the file had them. I definitely prefer to leave them out. I'll send an update shortly.

Reply:
> Well, for rare cases we add the fail messages as it helps debug.

That's kind of the point of using the specific matcher methods (assert.include, assert.truthy, assert.lengthOf etc) instead of using `strictEqual(customLogic, true, descriptionOfCustomLogic)`.

Reply:
@eps1lon You know my position on the topic. I'm all for using as simple assertion as possible, it's critical to have trust in them. There is nothing simpler than `strictEqual`. Now, it can be a pain to test everything with a `===`, testing an exception or a string inclusion are two great examples. I was lured at some point by power-assert, but I have never seen it working in real life.

Reply:
All good. Just wanted to take a cheap shot :smile: I think the main difference is between matcher and assert style. It makes more sense to use only primitive assertions over complex assertions. If I had to choose I would always go with `expect(myElement.children).toHaveLength(6)` over `assert.strictEqual(myElement.children.length, 6)`. Matcher style is closer to natural language and has some built-in guards if myElement.children doesn't have a `length` property. 

Takes some time to learn the vocabulary but it makes life so much easier if you work on an unknown code, change it, break test and understand what the test was about. `assert` is to technical for my taste. It boils down to preferring code that's easier to read but harder to write.